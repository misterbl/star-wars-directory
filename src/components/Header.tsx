import React, { PureComponent } from "react";
import { bindActionCreators, Dispatch } from "redux";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { Formik } from "formik";
import { IHeaderComponent } from "./Header.d";
import searchFilmsAndPeople from "../actions/thunks/searchFilmsAndPeople";
import SearchForm, { ISearchForm } from "./SearchForm";
import ROUTES from "../routes";

export class Header extends PureComponent<IHeaderComponent> {
  handleSubmit = async (data: ISearchForm) => {
    const {
      searchFilmsAndPeople,
      history: { push }
    } = this.props;
    await searchFilmsAndPeople(data.searchTerm);
    push(ROUTES.INDEX);
  };

  render() {
    return (
      <div className="header">
        <Formik<ISearchForm>
          initialValues={{
            searchTerm: ""
          }}
          onSubmit={this.handleSubmit}
          render={formikProps => <SearchForm {...formikProps} />}
        />
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch: Dispatch) => ({
  searchFilmsAndPeople: bindActionCreators(searchFilmsAndPeople, dispatch)
});

export default withRouter(
  connect(
    null,
    mapDispatchToProps
  )(Header)
);
