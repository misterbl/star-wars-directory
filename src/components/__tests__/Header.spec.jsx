/* eslint-disable no-undef */
import * as React from "react";
import { shallow } from "enzyme";
import history from "../../testMocks/history.mock";
import { Header } from "../Header";
import ROUTES from "../../routes";

describe("Header", () => {
  const props = {
    history,
    searchFilmsAndPeople: jest.fn()
  };
  const wrapper = shallow(<Header {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("calls searchFilmsAndPeople() Formik onSubmit ", () => {
    const mainDiv = wrapper.find("Formik");
    mainDiv.simulate("submit", { searchTerm: "searchTerm" });
    expect(props.searchFilmsAndPeople).toHaveBeenCalled();
  });
  it("calls push() Formik onSubmit ", () => {
    const mainDiv = wrapper.find("div").at(0);
    mainDiv.simulate("click");
    expect(props.history.push).toHaveBeenCalledWith(ROUTES.INDEX);
  });
});
