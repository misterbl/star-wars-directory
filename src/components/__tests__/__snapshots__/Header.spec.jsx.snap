// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    history={
      Object {
        "action": "PUSH",
        "block": [MockFunction],
        "createHref": [MockFunction],
        "go": [MockFunction],
        "goBack": [MockFunction],
        "goForward": [MockFunction],
        "length": 123,
        "listen": [MockFunction],
        "location": Object {
          "hash": "",
          "key": "",
          "pathname": "",
          "search": "",
          "state": "",
        },
        "push": [MockFunction],
        "replace": [MockFunction],
      }
    }
    searchFilmsAndPeople={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="form-group mb-0 search-form search-form__header"
      >
        <Formik
          enableReinitialize={false}
          initialValues={
            Object {
              "searchTerm": "",
            }
          }
          isInitialValid={false}
          onSubmit={[Function]}
          render={[Function]}
          validateOnBlur={true}
          validateOnChange={true}
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Formik
          enableReinitialize={false}
          initialValues={
            Object {
              "searchTerm": "",
            }
          }
          isInitialValid={false}
          onSubmit={[Function]}
          render={[Function]}
          validateOnBlur={true}
          validateOnChange={true}
        />,
        "className": "form-group mb-0 search-form search-form__header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "enableReinitialize": false,
          "initialValues": Object {
            "searchTerm": "",
          },
          "isInitialValid": false,
          "onSubmit": [Function],
          "render": [Function],
          "validateOnBlur": true,
          "validateOnChange": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="form-group mb-0 search-form search-form__header"
        >
          <Formik
            enableReinitialize={false}
            initialValues={
              Object {
                "searchTerm": "",
              }
            }
            isInitialValid={false}
            onSubmit={[Function]}
            render={[Function]}
            validateOnBlur={true}
            validateOnChange={true}
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Formik
            enableReinitialize={false}
            initialValues={
              Object {
                "searchTerm": "",
              }
            }
            isInitialValid={false}
            onSubmit={[Function]}
            render={[Function]}
            validateOnBlur={true}
            validateOnChange={true}
          />,
          "className": "form-group mb-0 search-form search-form__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "enableReinitialize": false,
            "initialValues": Object {
              "searchTerm": "",
            },
            "isInitialValid": false,
            "onSubmit": [Function],
            "render": [Function],
            "validateOnBlur": true,
            "validateOnChange": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
